### PR TITLE
Add (preliminary) Server Routes

### PR DESCRIPTION
**Pull request by addisonhernandez**
I added some routes to the models and controllers, based on the Books API and Book Clubs API endpoints.

I think we need to make some changes to deconflict some of the routes? Let me know what you think

### CODE REVIEW

File: server/routes.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+const controller = require('./controllers');
+const router = require('express').Router();
+
+// ---                     ---
+// --- Books API endpoints ---
+// ---                     ---
+router.get('/search', controller.books.searchBooks);
+router.get('/books', controller.books.personalBooks); // FIXME: conflict with react-router endpoint
```
Comment (katyfsy) [inline server/routes.js:8]: React-router endpoint doesn't seem to have `'/books'` route, seeing a `'/mybooks'` route instead, can we double confirm?

File: server/routes.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+router.get('/popularBooks', controller.books.popularBooks);
+
+router.post('/books', controller.books.addBook); // FIXME: conflict with react-router endpoint
```
Comment (katyfsy) [inline server/routes.js:11]: React-router endpoint doesn't seem to have `'/books'` route, seeing a `'/mybooks'` route instead, can we double confirm?

File: server/routes.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+
+// FIXME: These need unique endpoints
+router.put('/books', controller.books.updateStatus);
```
Comment (katyfsy) [inline server/routes.js:14]: updated route:` '/books/status'`

File: server/routes.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+router.put('/books', controller.books.addToBookshelf);
```
Comment (katyfsy) [inline server/routes.js:15]: updated route: `'/books/bookshelf'`

File: server/routes.js [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+router.put('/books', controller.books.reviewBook);
```
Comment (katyfsy) [inline server/routes.js:16]: updated route: `'/books/review'`

File: server/routes.js [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+router.put('/books', controller.books.rateBook);
```
Comment (katyfsy) [inline server/routes.js:17]: updated route:` '/books/rating'`

File: server/routes.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
...
+
+// ---                          ---
+// --- Book Clubs API endpoints ---
+// ---                          ---
+router.get('/bookclubs', controller.bookclubs.allClubs);
+router.get('/mybookclubs', controller.bookclubs.myClubs);
+
+router.post('/bookclub', controller.bookclubs.createClub);
```
Comment (katyfsy) [inline server/routes.js:25]: updated route:` '/bookclubs/create'`

File: server/routes.js [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+router.post('/FIXME', controller.bookclubs.postMessage); // FIXME: API endpoint not yet defined
```
Comment (katyfsy) [inline server/routes.js:26]: updated route: `'/bookclubs/messages'`

File: server/routes.js [hunk 9]
```diff
@@ -0,0 +1,30 @@
...
+
+router.put('/FIXME', controller.bookclubs.joinClub); // FIXME: API endpoint not yet defined
```
Comment (katyfsy) [inline server/routes.js:28]: updated route: `'/bookclubs/join'`

File: server/routes.js (hunk 1 above)
Comment (addisonhernandez) [reply server/routes.js:8]: :100: Good call!

That's an easy fix. I'll just remove the comment :+1:

**Comment by addisonhernandez** [comment]
[Server-side routes work-in-progress proposal](https://trello.com/c/oro4mwwJ/8-server-side-routes-work-in-progress-proposal)

**Review by HuijunLu — APPROVED** [review]
Looks good. I added the one endpoint for deleting bookclubs and updated the mybookclubs endpoint to myBookclubs.

**Pull request merged**